pkg/translator/prometheusremotewrite/metrics_to_prw_v2.go

```diff
@@ -107,16 +108,24 @@ func (c *prometheusConverterV2) timeSeries() []writev2.TimeSeries {
 	return allTS
 }
 
-func (c *prometheusConverterV2) addSample(sample *writev2.Sample, lbls labels.Labels) *writev2.TimeSeries {
+func (c *prometheusConverterV2) addSample(sample *writev2.Sample, lbls []prompb.Label) *writev2.TimeSeries {
 	if sample == nil || len(lbls) == 0 {
 		// This shouldn't happen
 		return nil
 	}
-	ts := &writev2.TimeSeries{}
-	ts.LabelsRefs = c.symbolTable.SymbolizeLabels(lbls, ts.LabelsRefs)
-	ts.Samples = append(ts.Samples, *sample)
 
-	c.unique[lbls.Hash()] = ts
+	buf := make([]uint32, 0, len(lbls)*2)
+	for _, l := range lbls {
+		off := c.symbolTable.Symbolize(l.Name)
```

Comment:
I'm not sure how Go works here, but are we allocating `off` on every loop? Or does the compiler optimizes that? Maybe we want to declare `off` outside?

Reply:
done.